examples/meta/src/model_selection/tree_builder.sg

```diff
@@ -0,0 +1,24 @@
+#![create_kernel]
+Kernel k1 = kernel("GaussianKernel")
+Kernel k2 = kernel("PolyKernel")
+#![create_kernel]
+
+#![create_machine]
+Machine svm = machine("LibSVM")
+#![create_machine]
+
+#![create_master_node]
+ParameterNode node = new ParameterNode(svm)
+#![create_master_node]
+
+#![create_child_nodes]
+ParameterNode k1_node = new ParameterNode(k1)
+ParameterNode k2_node = new ParameterNode(k2)
+#![create_child_nodes]
+
+#![attach_child_nodes]
+k2_node.attach("degree", 2)
+node.attach("kernel", k1_node)
```

Comment:
I am just having the crazy idea of re-using `put/add` here: 
The constructor `new ParameterNode(k1)` returns an object that has parameters registered inside (it checks which parameters the provided SGObject has and registers the same in the returned object). Then users just would do `k1_node.add("kernel", 2)` for adding to the options, or `k1_node.put("kernel", 2)` if they want a single value?
I am not sure whether it is really a good idea to do this since it has other implications, but it would save you the pain from implementing "attach" and adding all the typed versions for it, which is kinda re-implementing a lot of stuff in the swig interface (e.g. the java hacks). Thoughts?